src/Symfony/Component/Serializer/Mapping/AttributeMetadata.php

```diff
@@ -42,13 +42,13 @@ class AttributeMetadata implements AttributeMetadataInterface
     public $maxDepth;
 
     /**
-     * @var string|null
+     * @var array<string, string|null> An array of serialized names by group
      *
      * @internal This property is public in order to reduce the size of the
      *           class' serialized representation. Do not access it. Use
-     *           {@link getSerializedName()} instead.
+     *           {@link getSerializedNames()} instead.
      */
-    public $serializedName;
+    public $serializedName = [];
```

Comment:
For maximum BC, I think we should keep this as a string when possible, and use arrays only when actually required.

Reply:
Tu be sure 

````
/**
* string|array<string, string|null>
*/
public $serializedName;
````

And deal with string and array in every methods, is it right ?